### PR TITLE
Port DeepLabV3 to Keras Core

### PR DESCRIPTION
**Pull request by ianstenbit**
I had to make some small adjustments to the model to a) clean up the docs a bit and b) get it in line with the paper
(https://arxiv.org/pdf/1802.02611.pdf)

### CODE REVIEW

File: keras_cv/layers/spatial_pyramid.py [hunk 1]
```diff
@@ -142,21 +144,28 @@ def call(self, inputs, training=None):
...
-        result = tf.concat(result, axis=-1)
+        result[-1] = keras.layers.Resizing(
+            inputs.shape[1], inputs.shape[2], interpolation="bilinear"
+        )(result[-1])
+
+        result = ops.concatenate(result, axis=-1)
         result = self.projection(result, training=training)
         return result
 
+    def compute_output_shape(self, input_shape):
```
Comment (jbischof) [inline keras_cv/layers/spatial_pyramid.py:166]: Why is this method needed in Keras Core?

File: keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py [hunk 2]
```diff
@@ -102,13 +107,20 @@ def __init__(
 
         inputs = backbone.input
 
-        backbone_outputs = backbone(inputs)
+        high_level_feature_extractor = keras.Model(
```
Comment (jbischof) [inline keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py:110]: Can't we use `get_feature_extractor` as in `RetinaNet` ([link](https://github.com/keras-team/keras-cv/blob/b1888aac446e6e49c0bb2bd9ac33586f27eabee5/keras_cv/models/object_detection/retinanet/retinanet.py#L174))?

**Review by jbischof — APPROVED** [review]
Thanks!

File: keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py [hunk 3]
```diff
@@ -102,13 +107,20 @@ def __init__(
...
+        high_level_features = high_level_feature_extractor(inputs)
 
         if spatial_pyramid_pooling is None:
             spatial_pyramid_pooling = SpatialPyramidPooling(
                 dilation_rates=[6, 12, 18]
             )
-        spp_outputs = spatial_pyramid_pooling(backbone_outputs)
+        spp_outputs = spatial_pyramid_pooling(high_level_features)
 
         low_level_feature_extractor = keras.Model(
```
Comment (jbischof) [inline keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py:125]: Same here

File: keras_cv/layers/spatial_pyramid.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/layers/spatial_pyramid.py:166]: It's needed for tracing the shape while tracing the forward pass. Without it we get an error like this one:
```
Could not automatically infer the output shape / dtype of 'spatial_pyramid_pooling_9' (of type SpatialPyramidPooling). 
Either the `SpatialPyramidPooling.call()` method is incorrect, or you need to implement the `SpatialPyramidPooling.compute_output_spec() / compute_output_shape()` method. Error encountered:
```

File: keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py:110]: We can, and we should! Done

**Pull request merged**